### DISCUSSION TITLE
Select discrete or integrated gpu

### DISCUSSION THREAD

**Discussion by filipforsstrom** [bevyengine/bevy#8851] [Q&A]
I'm new to developing applications with graphics and currently building a midi sequencer with Bevy. Is it possible to select which gpu bevy should use? When I run the app my AdapterInfo is:

```
INFO bevy_render::renderer: AdapterInfo { name: "AMD Radeon Pro 5500M", vendor: 0, device: 0, device_type: DiscreteGpu, driver: "", driver_info: "", backend: Metal }
```

I would like to change to the integrated gpu and see if I get the same performance but with lower power consumption. I have two computers - one macbook pro 2017 without a discrete gpu and intel i7 and the other macbook pro 2019 with a AMD Radeon Pro 5500M and intel i9. I've noticed that the fans are running much faster on the one with a discrete gpu which I can understand. But I don't understand why the cpu usage is much higher on the macbook pro 2019.

**Comment by rlidwka** [comment]
You can switch between integrated GPU and discrete GPU by setting power preference in wgpu settings:

```rs
app.add_plugins(DefaultPlugins
    .set(RenderPlugin {
        wgpu_settings: WgpuSettings {
            power_preference: PowerPreference::LowPower,
            ..default()
        }
    })
);
```

But I haven't found any way to select specific adapter (if you have 3 of them for example). You can list all available adapters with `enumerate_adapters` (https://github.com/gfx-rs/wgpu/discussions/2022), but there's no way to select specific one to use in bevy?

**Comment by decepticlown** [comment]
Currently NVidia drivers are broken with Vulkan on Wayland. I have to use `env VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/radeon_icd.x86_64.json cargo run` to run app otherwise with NVidia Vulkan window freezes permanently and won't accept any keys. Even ctrl-c/ctrl-d doesn't work. So having option to choose adapter will be really helpful if client is also using NVidia with Wayland.

**Comment by decepticlown** [comment]
By the way in bevy 0.13 you can do low power mode with
```
            DefaultPlugins.set(RenderPlugin {
                render_creation: WgpuSettings {
                    power_preference: PowerPreference::LowPower,
                    ..Default::default()
                }
                .into(),
                ..Default::default()
            }),
```
